pkg/instancetype/instancetype.go

```diff
@@ -357,6 +365,67 @@ func getInstancetypeAPISpec(obj runtime.Object) (interface{}, error) {
 	}
 }
 
+func checkCPUPreferenceRequirements(instancetypeSpec *instancetypev1alpha2.VirtualMachineInstancetypeSpec, preferenceSpec *instancetypev1alpha2.VirtualMachinePreferenceSpec, vmiSpec *virtv1.VirtualMachineInstanceSpec) (*k8sfield.Path, error) {
+	if instancetypeSpec != nil && instancetypeSpec.CPU.Guest < preferenceSpec.Requirements.CPU.Guest {
```

Comment:
Move `instancetypeSpec == nil` at the beginning of the function and let it return early if it is `nil` instead of checking several times?

Reply:
Meh, I'd still have a conditional for the other half of these checks so returning early doesn't really help IMHO.